docs/containers-storage-architecture.md

```diff
@@ -0,0 +1,76 @@
+## containers-storage 1 "May 2024"
+
+## NAME
+containers-storage-architecture - The implementation of containers-storage 
+
+NOTE: These are implementation details subject to change. This document
+is intended to be used only as an aid for debugging and development
+of the containers-storage system. Do *not* directly access the filesystem
+components below from external code, only from the Go API.
+
+## DESCRIPTION
+The *containers/storage* library manages the following object types:
+
+- layer: A filesystem tree, along with metadata
+- image: A reference to a layer, along with metadata
+- container: A layer whose parent is an image
+
+## Layer storage
+
+### `overlay/`
```

Comment:
I, very weakly, think that pointing at, and improving, the documentation in `drivers/overlay/overlay.go` might be more helpful than duplicating it.

Reply:
Yes, this makes total sense.  I've moved this text into `overlay.go.`